hazelcast/src/test/java/com/hazelcast/internal/jmx/LocalStatsDelegateTest.java

```diff
@@ -16,168 +16,133 @@
 public class LocalStatsDelegateTest extends HazelcastTestSupport {
 
     private AtomicBoolean done;
-    private HazelcastInstance hz;
 
     @Before
     public void setUp() {
-        hz = createHazelcastInstance();
         done = new AtomicBoolean(false);
     }
 
     @Test
-    public void testMapStats_Created_OnlyOnce_InGivenInternal() throws InterruptedException {
-        final IMap<Object, Object> trial = hz.getMap("trial");
-
-        final StatsSupplier<LocalMapStats> statsSupplier = new LocalMapStatsSupplier(trial);
-        final LocalStatsDelegate localStatsDelegate = new LocalStatsDelegate<LocalMapStats>(statsSupplier, 60);
-
-        Thread mapStatsThread1 = new Thread(new MapStatsThread(localStatsDelegate, done, 100));
-        Thread mapStatsThread2 = new Thread(new MapStatsThread(localStatsDelegate, done, 90));
-
-        mapStatsThread1.start();
-        mapStatsThread2.start();
-
-        Thread mapPutThread = new Thread(new MapPutThread(trial, done));
-        mapPutThread.start();
-
-        sleepSeconds(30);
-
-        done.set(true);
-
-        mapStatsThread1.join();
-        mapStatsThread2.join();
-        mapPutThread.join();
+    @Category(NightlyTest.class)
+    public void testMapStats_Created_OnlyOnce_InGivenInternal() throws Exception {
+        test(false, 60, 100, 90, 30);
     }
 
     /**
-     * In this test, map entry count is incrementing steadily. Thus, we expect OwnedEntryCount
-     * to increase as well. Since the interval value for generating new map stats is 5 seconds,
-     * we assert for greater or equal.
+     * In this test, map entry count is incrementing steadily. Thus, we expect OwnedEntryCount to increase as well. Since the
+     * interval value for generating new map stats is 5 seconds, we assert for greater or equal.
      */
     @Test
     @Category(NightlyTest.class)
-    public void stressTest() throws InterruptedException {
-        final IMap<Object, Object> trial = hz.getMap("trial");
+    public void stressTest() throws Exception {
+        test(true, 5, 100, 80, 60);
+    }
+
+    private void test(final boolean stress, final int intervalSec, final int mapStatsThread1SleepMs,
+            final int mapStatsThread2SleepMs, final int sleepSeconds) throws Exception {
+        final IMap<Object, Object> trial = createHazelcastInstance().getMap("trial");
 
         final StatsSupplier<LocalMapStats> statsSupplier = new LocalMapStatsSupplier(trial);
-        final LocalStatsDelegate localStatsDelegate = new LocalStatsDelegate<LocalMapStats>(statsSupplier, 5);
+        final LocalStatsDelegate<LocalMapStats> localStatsDelegate = new LocalStatsDelegate<>(statsSupplier, intervalSec);
 
-        Thread mapStatsThread1 = new Thread(new MapStatsThread(localStatsDelegate, done, true, 100));
-        Thread mapStatsThread2 = new Thread(new MapStatsThread(localStatsDelegate, done, true, 80));
+        final ExecutorService executor = Executors.newFixedThreadPool(3);
```

Comment:
The `executor` is not shutdown after the test so resources will leak. If you use the executor, make it a field in the test class so you can shut it down in `@AfterClass`. Alternatively, you can use `HazelcastTestSupport#spawn` to conveniently spawn threads.